indigo_api/migrations/0125_migrate_commencements.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.27 on 2020-01-09 14:59
+from __future__ import unicode_literals
+
+import re
+
+from django.conf import settings
+from django.db import migrations
+
+from indigo_api.models import Act
+from indigo.plugins import plugins
+
+
+def migrate_commencements(apps, schema_editor):
+    """ Copy commencement details from works to Commencement objects
+    """
+    Work = apps.get_model("indigo_api", "Work")
+    Document = apps.get_model("indigo_api", "Document")
+    Commencement = apps.get_model("indigo_api", "Commencement")
+    db_alias = schema_editor.connection.alias
+
+    def for_document(topic, document):
+        country_code = document.work.country.country.iso.lower()
+        locality = document.work.locality.code if document.work.locality else None
+        language_code = document.language.language.iso_639_2B
+        return plugins.for_locale(topic, country=country_code, locality=locality, language=language_code)
+
+    def table_of_contents_for_document(self, document):
+        def make_act(xml):
+            id = re.sub(r'[^a-zA-Z0-9]', '-', settings.INDIGO_ORGANISATION)
```

Comment:
this `id` stuff isn't necessary for the TOC, so you can drop it.

Reply:
don't I need it to make `doc` so that `.table_of_contents()` on l36 works as expected?

Reply:
nvm, just read the next comment